library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -149,7 +149,7 @@ public void onCreate() {
                 Downloads.Impl.ALL_DOWNLOADS_CONTENT_URI,
                 true, mObserver);
 
-        mExecutor = new DownloadExecutorFactory().createExecutor(this);
+        mExecutor = DownloadExecutorFactory.newInstance(this).createExecutor();
```

Comment:
just inline this newInstance method, can't see it adding anything (lets not refactor prematurely)